### PR TITLE
Subscription sidebar

### PR DESCRIPTION
**Pull request by BentiGorlich**
Right now the sidebar is very basic. It just shows your subscriptions by their last post. If you have more than 50 subscriptions it show the first 50 and adds a "more" button underneath where you can get to your subscriptions. 

I think we can get the basic version out there and collect feedback as to what should be added or changed about it.

There might still be some rebase bugs in it as I had to rebase this code about 5 times by now, but hopefully I catched them all.

#138

### CODE REVIEW

**Review by TheVillageGuy — APPROVED** [review]

**Review by nobodyatroot — DISMISSED** [review]
Pretty spiffy, I like it! As discussed in matrix chat, it might be a nice addon or different PR to give the user the ability to turn off the random junk and replace all the content with their subscribed stuff, even "random" posts/threads/etc _from_ their subscriptions. I think that'd be pretty useful/insightful.

**Comment by BentiGorlich** [comment]
I am not sure about the positioning of the subscription panel, I think if you're in a magazine the subscription panel shouldn't be on top of it as it is now, but I don't know. Thats up for debate

**Comment by BentiGorlich** [comment]
> Pretty spiffy, I like it! As discussed in matrix chat, it might be a nice addon or different PR to give the user the ability to turn off the random junk and replace all the content with their subscribed stuff, even "random" posts/threads/etc _from_ their subscriptions. I think that'd be pretty useful/insightful.

I think that should be a separate issue :D

**Comment by BentiGorlich** [comment]
Sorry  @TheVillageGuy @nobodyatroot  I added a commit, I promise this was the last one :D

File: src/Controller/User/ThemeSettingsController.php [hunk 1]
```diff
@@ -109,6 +119,10 @@ public function __invoke(string $key, string $value, Request $request): Response
             $response->headers->setCookie(new Cookie($key, $value, strtotime('+1 year')));
         }
 
+        //        if (self::KBIN_THEME === $key && self::KBIN === $value) {
```
Comment (e-five256) [inline src/Controller/User/ThemeSettingsController.php:122]: could these lines be cleaned up?

File: templates/layout/_options_appearance.html.twig [hunk 2]
```diff
@@ -10,10 +10,25 @@
         {% if kbin_mercure_enabled() %}
             {{ component('settings_row_switch', {label: 'dynamic_lists'|trans, settingsKey: 'KBIN_GENERAL_DYNAMIC_LISTS'}) }}
         {% endif %}
-        {{ component('settings_row_switch', {label: 'rounded_edges'|trans, settingsKey: 'KBIN_GENERAL_ROUNDED_EDGES', defaultValue: true}) }}
+        {{ component('settings_row_enum', {label: 'sidebar_position'|trans, settingsKey: 'KBIN_GENERAL_SIDEBAR_POSITION', values: [ {name: 'left'|trans , value: 'LEFT'}, {name: 'right'|trans , value: 'RIGHT' } ], defaultValue: 'RIGHT' } ) }}
```
Comment (e-five256) [inline templates/layout/_options_appearance.html.twig:13]: was this meant to be duplicated? there are two sidebar position entries now

File: templates/layout/_options_appearance.html.twig [hunk 3]
```diff
@@ -10,10 +10,25 @@
...
+        {{ component('settings_row_switch', {label: 'rounded_edges'|trans, settingsKey: 'KBIN_GENERAL_ROUNDED_EDGES'}) }}
```
Comment (e-five256) [inline templates/layout/_options_appearance.html.twig:14]: it looks like this lost the default value setting, not sure if that was on purpose

**Comment by e-five256** [comment]
if you turn on subscriptions separate and then turn off show subscriptions, you stay in the 3 column setup but can't go back without turning show subscriptions back on. :thinking: not sure if there's an easy way to say if one is turned off turn the other off as I'm not super familiar with it

**Comment by BentiGorlich** [comment]
applied your feedback @e-five256

**Review by nobodyatroot — DISMISSED** [review]
Re-approving! :-)

**Comment by e-five256** [comment]
Was the intention that the sub name gets hidden at lower resolutions? I took to code to mean hide the avatar, but it seems to do the name

image

**Review by e-five256 — DISMISSED** [review]

**Comment by BentiGorlich** [comment]
Sorry, gotta add another commit. The mobile view is broken. Already fixed it, but you need to reapprove it :sweat_smile:

**Comment by BentiGorlich** [comment]
grafik

**Review by e-five256 — APPROVED** [review]

**Comment by melroy89** [comment]
.. Let's see what weblate think about this change.

**Comment by melroy89** [comment]
> .. Let's see what weblate thinks about this change.

Sync seems to work.

**Comment by melroy89** [comment]
@BentiGorlich some feedback still, despite it being merged already.. "Sort" is unclear it's about sorting of the subscriptions, please rename this to make it explicit:

image

**Pull request merged**